util/patch/patch_test.go

```diff
@@ -997,3 +999,1683 @@ func TestNewHelperNil(t *testing.T) {
+				if err != nil || !ok {
+					return false
+				}
+
+				readyBefore := conditions.Get(obj, clusterv1.ReadyCondition)
+				readyAfter := conditions.Get(objAfter, clusterv1.ReadyCondition)
+				ok, err = conditions.MatchCondition(*readyBefore).Match(*readyAfter)
+				if err != nil || !ok {
+					return false
+				}
+
+				return obj.Spec.Foo == objAfter.Spec.Foo &&
+					obj.Status.Bar == objAfter.Status.Bar
+			}, timeout).Should(BeTrue(), cmp.Diff(obj, objAfter))
+		})
+
+		t.Run("should return an error if there is an unresolvable conflict", func(t *testing.T) {
+			g := NewWithT(t)
+
+			obj := obj.DeepCopy()
+
+			t.Log("Creating the object")
+			g.Expect(env.Create(ctx, obj)).To(Succeed())
+			defer func() {
+				g.Expect(env.Delete(ctx, obj)).To(Succeed())
+			}()
+			key := client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}
+
+			t.Log("Checking that the object has been created")
+			g.Eventually(func() error {
+				obj := obj.DeepCopy()
+				return env.Get(ctx, key, obj)
+			}).Should(Succeed())
+
+			objCopy := obj.DeepCopy()
+
+			t.Log("Marking a custom condition to be false")
+			conditions.MarkFalse(objCopy, clusterv1.ReadyCondition, "reason", clusterv1.ConditionSeverityInfo, "message")
+			g.Expect(env.Status().Update(ctx, objCopy)).To(Succeed())
+
+			t.Log("Validating that the local object's resource version is behind")
+			g.Expect(obj.ResourceVersion).NotTo(Equal(objCopy.ResourceVersion))
+
+			t.Log("Creating a new patch helper")
+			patcher, err := NewHelper(obj, env)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			t.Log("Marking Ready=True")
+			conditions.MarkTrue(obj, clusterv1.ReadyCondition)
+
+			t.Log("Patching the object")
+			g.Expect(patcher.Patch(ctx, obj)).NotTo(Succeed())
+
+			t.Log("Validating the object has not been updated")
+			g.Eventually(func() bool {
+				objAfter := obj.DeepCopy()
+				if err := env.Get(ctx, key, objAfter); err != nil {
+					return false
+				}
+
+				for _, afterCondition := range objAfter.Status.Conditions {
+					ok, err := conditions.MatchCondition(objCopy.Status.Conditions[0]).Match(afterCondition)
+					if err == nil && ok {
+						return true
+					}
+				}
+
+				return false
+			}, timeout).Should(BeTrue())
+		})
+
+		t.Run("should not return an error if there is an unresolvable conflict but the conditions is owned by the controller", func(t *testing.T) {
+			g := NewWithT(t)
+
+			obj := obj.DeepCopy()
+
+			t.Log("Creating the object")
+			g.Expect(env.Create(ctx, obj)).To(Succeed())
+			defer func() {
+				g.Expect(env.Delete(ctx, obj)).To(Succeed())
+			}()
+			key := client.ObjectKey{Name: obj.Name, Namespace: obj.Namespace}
+
+			t.Log("Checking that the object has been created")
+			g.Eventually(func() error {
+				obj := obj.DeepCopy()
+				return env.Get(ctx, key, obj)
+			}).Should(Succeed())
+
+			objCopy := obj.DeepCopy()
+
+			t.Log("Marking a custom condition to be false")
```

Comment:
```suggestion
			t.Log("Marking Ready=False")
```
Same in l.1302 and probably a lot more occurences, please check for other cases below

Reply:
Fixed all around